table/block_based/full_filter_block.cc

```diff
@@ -245,7 +245,7 @@ void FullFilterBlockReader::MayMatch(
   MultiGetRange filter_range(*range, range->begin(), range->end());
   for (auto iter = filter_range.begin(); iter != filter_range.end(); ++iter) {
     if (!prefix_extractor) {
-      keys[num_keys++] = &iter->ukey;
+      keys[num_keys++] = &iter->ukey_without_ts;
     } else if (prefix_extractor->InDomain(iter->ukey)) {
```

Comment:
Aren't all of these wrong?

    $ git grep -- '->ukey' | grep filter
    table/block_based/filter_block.h:      const Slice ukey = iter->ukey;
    table/block_based/filter_block.h:      const Slice ukey = iter->ukey;
    table/block_based/full_filter_block.cc:      keys[num_keys++] = &iter->ukey;
    table/block_based/full_filter_block.cc:    } else if (prefix_extractor->InDomain(iter->ukey)) {
    table/block_based/full_filter_block.cc:      prefixes.emplace_back(prefix_extractor->Transform(iter->ukey));
    $

Reply:
hmm, I think prefix is a prefix of the user key, thus should be shorter than or equal to user key. The obtained prefix from a user key should not contain timestamp.

Reply:
In fact, I'm not really seeing instances of ->ukey that I would suspect *not* to be impacted by user timestamp. Perhaps we should rename `ukey` to `ukey_with_ts`. (I'm not saying you have to fix everything in this PR.)

Reply:
> hmm, I think prefix is a prefix of the user key, thus should be shorter than or equal to user key. The obtained prefix from a user key should not contain timestamp.

I'm not aware of anything that guarantees that. AFAIK, prefix extractor can be identity function.

Reply:
If so, we need to update the code to make sure prefix extractor operates only on user key without ts.

Reply:
Updated, and added test for that.